generator/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"math"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+type Event struct {
+	Status int
+}
+
+func setInterval(function func(), seconds int) {
+	ticker := time.NewTicker(time.Duration(seconds) * time.Second)
+	done := make(chan bool)
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				function()
+			case <-done:
+				ticker.Stop()
+				return
+			}
+		}
+	}()
+}
+
+func main() {
+	setInterval(GenerateWindEvents, 5)
+	http.ListenAndServe(":1234", nil)
+}
+
+func GenerateWindEvents() {
+	if checkForStatus() != 1 {
```

Comment:
Manage error here, not only status, you should be able to see what do you get from your response and log it at least